### DISCUSSION TITLE
PKCS12_create in 3.4 fips mode behavior

### DISCUSSION THREAD

**Discussion by ChauhanPriyanka** [openssl/openssl#25895] [Q&A] [answered]
Hi Team,

As per the release notes - https://github.com/openssl/openssl/blob/openssl-3.4.0/NEWS.md. 
The implementation of RFC 9579 (PBMAC1) in PKCS#12 is completed, does this mean the we will now be able to create PKCS12 structure via PKCS12_create api in fips mode. i.e. both PKCS12_create and PKCS12_parse API will work in fips mode with 3.4 release.

Reference ticket for the behavior is - https://github.com/openssl/openssl/issues/19997

Thanks,
Priyanka.

**Comment by beldmit** [comment, marked as answer]
Yes, you can see how it is implemented in the `pkcs12` util

**Comment by StephenWall** [comment]
Any idea what (certified) version of the FIPS provider will this arrive in?  Is it realistic to port this to a 3.0.x provider?

**Reply by t8m** [reply]
AFAIK you can get this feature if you use an existing validated version of FIPS provider with the current 3.4.0 release of libcrypto/libssl.
